lib/api/_loaders/element-global.js

```diff
@@ -80,14 +81,30 @@ class ElementGlobal {
       return;
     }
 
-    const {locator} = this;
+    let {locator} = this;
     if ((locator instanceof Element) && locator.resolvedElement) {
       this.element = this.createWebElement(locator.resolvedElement);
 
       return;
     }
 
-    this.element = await this.transport.driver.wait(until.elementLocated(locator), this.timeout, null, this.retryInterval);
+    if (locator instanceof Element) {
+      this.setPropertiesFromElement(locator);
+
+      locator = Locator.create(locator);
```

Comment:
There are multiple calls to `Locator.create`. We should refactor this module and everything should be handled in `setLocator` method.